server/src/main/java/io/deephaven/server/barrage/BarrageStreamGenerator.java

```diff
@@ -473,14 +481,13 @@ public SnapshotView(final BarrageStreamGenerator generator,
 
             // precompute add row offsets
             if (keyspaceViewport != null) {
-                try (WritableRowSet intersect = keyspaceViewport.intersect(generator.rowsIncluded.original)) {
-                    addRowOffsets = generator.rowsIncluded.original.invert(intersect);
-                }
+                addRowKeys = keyspaceViewport.intersect(generator.rowsIncluded.original);
+                addRowOffsets = generator.rowsIncluded.original.invert(addRowKeys);
             } else {
+                addRowKeys = generator.rowsAdded.original.copy();
                 addRowOffsets = RowSetFactory.flat(generator.rowsAdded.original.size());
```

Comment:
I liked the above simplification, which can be done here, too.

```suggestion
                addRowOffsets = RowSetFactory.flat(addRowKeys.size());
```